engine/table/src/main/java/io/deephaven/engine/util/BigDecimalUtils.java

```diff
@@ -62,18 +64,23 @@ public static PrecisionAndScale computePrecisionAndScale(
     public static PrecisionAndScale computePrecisionAndScale(
             final RowSet rowSet,
             final ColumnSource<BigDecimal> source) {
-        final int sz = 4096;
+        if (rowSet.isEmpty()) {
+            return EMPTY_TABLE_PRECISION_AND_SCALE;
+        }
+
         // we first compute max(precision - scale) and max(scale), which corresponds to
         // max(digits left of the decimal point), max(digits right of the decimal point).
         // Then we convert to (precision, scale) before returning.
         int maxPrecisionMinusScale = 0;
         int maxScale = 0;
-        try (final ChunkSource.GetContext context = source.makeGetContext(sz);
```

Comment:
Can we modify the block comment right before this point to make it explicit that we are not reading the whole table, just making an estimate based on the first PRECISION_SCALE_N_TO_INSPECT values?